libs/ds18b20/ds18b20.cpp

```diff
@@ -1,19 +1,91 @@
 #include "ds18b20.h"
 
+/**
+ * Since neither DallasTemperature functions 'isConnected' nor
+ * 'isConversionComplete' work in parasite mode, conversion time has to
+ * be considere explicitly.
+ * 9-12 bit resolution: 94, 188, 375, 750 ms conversion time
+ */
+#define DS18B20_RESOLUTION 10 // set desired resolution in bit (9 to 12)
+#define DS18B20_CONVERSION_DURATION 188
+
 Ds18b20::Ds18b20(uint8_t pin)
 : m_wire{pin}
 , m_sensors{&m_wire}
 {
+}
+
+Ds18b20::Ds18b20(uint8_t pin, uint8_t pin_pullup)
+: m_pin_pullup{pin_pullup}
+, m_wire{pin}
+, m_sensors{&m_wire}
+{
+    pinMode(m_pin_pullup, OUTPUT);
+    set_external_pullup(true);
+}
+
+void Ds18b20::begin()
+{
+    reset();
+}
```

Comment:
That's pretty pointless. Just drop the `begin` method and call `reset` directly.